### PR TITLE
Add Roman to Integer solution and tests

### LINKED ISSUE #3: Problem 13 - Roman to Integer
Problem 13 - Roman to Integer

## Description
I would like to add a solution to the [13. Roman to Integer](https://leetcode.com/problems/roman-to-integer/) problem. The problem is about converting a Roman numeral to an integer. Roman numerals are represented by seven different symbols: I, V, X, L, C, D, and M.

The problem has the following constraints:
- 1 <= s.length <= 15
- s contains only the characters ('I', 'V', 'X', 'L', 'C', 'D', 'M').
- It is guaranteed that s is a valid Roman numeral in the range [1, 3999].

### Example 1:
Input: s = "III"
Output: 3
Explanation: III = 3.

### Example 2:
Input: s = "LVIII"
Output: 58
Explanation: L = 50, V= 5, III = 3.

### Example 3:
Input: s = "MCMXCIV"
Output: 1994
Explanation: M = 1000, CM = 900, XC = 90, and IV = 4.

## Task
- Implement a solution for the problem in Rust
- Add unit tests for the implemented solution
- Update the README.md file with a link to the problem and a brief description

### PR DESCRIPTION
**Pull request by DieHard073055**
## Implement Roman to Integer Solution

This PR adds a `Solution` struct and implementation for the Roman to Integer problem.

### Solution overview:
The solution involves iterating through the input Roman numeral string and converting each character to its corresponding integer value using the `roman_char_to_int` function. During the iteration, the `eval_roman_chars` function is used to compare the current Roman character with the next one. This helps to identify and handle the cases where subtraction is required, such as 'IV' and 'IX'. If subtraction is needed, the function returns a tuple containing a boolean value `true` (indicating to skip the next character) and the integer value resulting from the subtraction. Otherwise, the function returns `false` and the integer value of the current Roman character.

The `roman_to_int` function iterates through the input string, applying the `eval_roman_chars` function to determine whether to add or subtract the corresponding integer values. It keeps track of the total value in the `total` variable. Once the iteration is complete, the `total` value is returned as the integer representation of the input Roman numeral.

### Implementation details:
- A `roman_char_to_int` function that converts a single Roman character to its corresponding integer value.
- An `eval_roman_chars` function that evaluates two Roman characters and returns a tuple containing a boolean value to indicate whether to skip the next character and an integer value to be added to the total.
- A `roman_to_int` function that converts a given Roman numeral string to its integer value.

Additionally, tests have been added to verify the correctness of the solution.

### CODE REVIEW

File: roman-to-integer/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    fn roman_char_to_int(roman: &char) -> i32 {
+        match roman {
+            'I' => 1,
+            'V' => 5,
+            'X' => 10,
+            'L' => 50,
+            'C' => 100,
+            'D' => 500,
+            'M' => 1000,
+            _ => !unreachable!("There should'nt be any more roman characters"),
```
Comment (pamdayne) [inline roman-to-integer/src/lib.rs]: typo `shouldn't`

File: roman-to-integer/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        }
+    }
+    fn eval_roman_chars(next: i32, previous: i32) -> (bool, i32) {
+        // println!("eval roman char: next {:}, previous {:}", next, previous);
```
Comment (pamdayne) [inline roman-to-integer/src/lib.rs]: should we clean up these `println`'s? 
im guessing these are for debugging purposes.

**Review by pamdayne — APPROVED** [review]

File: roman-to-integer/src/lib.rs (hunk 1 above)
Comment (DieHard073055) [reply roman-to-integer/src/lib.rs]: Done

File: roman-to-integer/src/lib.rs (hunk 2 above)
Comment (DieHard073055) [reply roman-to-integer/src/lib.rs]: Done

**Pull request merged**